codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -196,21 +221,33 @@ private static int getConnVariableHeaderFlag(MqttConnectVariableHeader variableH
     }
 
     private static ByteBuf encodeConnAckMessage(
-            ByteBufAllocator byteBufAllocator,
+            ChannelHandlerContext ctx,
             MqttConnAckMessage message) {
-        ByteBuf buf = byteBufAllocator.buffer(4);
+        final MqttVersion mqttVersion = MqttCodecUtil.getMqttVersion(ctx);
+        ByteBuf propertiesBuf = encodePropertiesIfNeeded(mqttVersion,
+                ctx.alloc(),
+                message.variableHeader().properties());
+
+        ByteBuf buf = ctx.alloc().buffer(4 + propertiesBuf.readableBytes());
         buf.writeByte(getFixedHeaderByte1(message.fixedHeader()));
-        buf.writeByte(2);
+        writeVariableLengthInt(buf, 2 + propertiesBuf.readableBytes());
         buf.writeByte(message.variableHeader().isSessionPresent() ? 0x01 : 0x00);
         buf.writeByte(message.variableHeader().connectReturnCode().byteValue());
+        buf.writeBytes(propertiesBuf);
+        propertiesBuf.release();
```

Comment:
I wonder if we should better do the release in a finally block. Just in case if something goes wrong... otherwise we will leak memory